libpod/container_copy_linux.go

```diff
@@ -237,21 +237,32 @@ func (c *Container) joinMountAndExec(ctx context.Context, f func() error) error
 		}
 		defer mountFD.Close()
 
-		pidFD, err := getFD(PIDNS)
+		hostPidNS, err := c.hostPidNS()
 		if err != nil {
-			errChan <- err
+			errChan <- errors.Wrap(err, "checking hostPidNS")
 			return
 		}
-		defer pidFD.Close()
-		if err := unix.Unshare(unix.CLONE_NEWNS); err != nil {
-			errChan <- err
-			return
+		var pidFD *os.File
```

Comment:
Do we need all the unsharing and the limbo above if `hostPidNS == true`?  I wonder if we couldn't just do a:
```golang
if hostPidNS {
     errChan <- f()
     return
}
```

... or even do that before line 219 outside of the go routine.

Reply:
We still need to join the mount namespace?

Reply:
Absolutely. Thanks for clarifying!